.github/workflows/Tests.yaml

```diff
@@ -0,0 +1,228 @@
+      - name: Check files
+        run: |
+          cd $GITHUB_WORKSPACE
+          ./tools/CheckFiles.sh
+
+  # Lint the code that changed in a pull request with clang-tidy.
+  clang_tidy:
+    name: Clang-tidy
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    container:
+      image: sxscollaboration/spectrebuildenv:latest
+    strategy:
+      matrix:
+        build_type: [Debug, Release]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Fetch upstream/develop
+        # We only want to check files that have changed in this pull request,
+        # so we fetch its base branch and pass it to `clang-tidy-hash`.
+        # Typically, it is the `sxs-collaboration/spectre/develop` branch.
+        run: >
+          cd $GITHUB_WORKSPACE
+
+          git remote add upstream
+          https://github.com/${{ github.repository }}.git
+
+          git remote -v
+
+          git fetch upstream ${{ github.base_ref }}
+      - name: Configure with cmake
+        working-directory: /work
+        run: >
+          mkdir build && cd build
+
+          cmake
+          -D CMAKE_C_COMPILER=clang
+          -D CMAKE_CXX_COMPILER=clang++
+          -D CMAKE_Fortran_COMPILER=gfortran-8
+          -D CHARM_ROOT=/work/charm/multicore-linux64-clang
+          -D CMAKE_BUILD_TYPE=${{ matrix.build_type }}
+          -D DEBUG_SYMBOLS=OFF
+          -D BUILD_PYTHON_BINDINGS=ON
+          $GITHUB_WORKSPACE
+      - name: Check clang-tidy
+        working-directory: /work/build
+        run: |
+          make clang-tidy-hash HASH=upstream/${{ github.base_ref }}
+
+  # Build the documentation and check for problems.
+  doc_check:
+    name: Documentation
+    runs-on: ubuntu-latest
+    container:
+      image: sxscollaboration/spectrebuildenv:latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Configure with cmake
+        working-directory: /work
+        run: >
+          mkdir build && cd build
+
+          cmake
+          -D CMAKE_C_COMPILER=clang
+          -D CMAKE_CXX_COMPILER=clang++
+          -D CMAKE_Fortran_COMPILER=gfortran-8
+          -D CHARM_ROOT=/work/charm/multicore-linux64-clang
+          -D CMAKE_BUILD_TYPE=Debug
+          -D DEBUG_SYMBOLS=OFF
+          -D BUILD_PYTHON_BINDINGS=ON
+          $GITHUB_WORKSPACE
+      - name: Check documentation
+        working-directory: /work/build
+        # Append `texlive` to the `PATH` so bibliography generation works.
+        # Can be removed once this is done in the container.
+        run: |
+          PATH=$PATH:/work/texlive/bin/x86_64-linux make doc-check
+
+  # Build all test executables and run unit tests on a variety of compiler
+  # configurations.
+  unit_tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: [gcc-6, gcc-7, gcc-8, gcc-9, clang-4.0, clang-5.0, clang-8]
+        build_type: [Debug, Release]
```

Comment:
Note on this configuration: By default, the other jobs in this matrix are canceled if one of them fails (`fail-fast` option). This only applies to the jobs in this matrix, not the other jobs (e.g. the documentation check and clang-tidy will not be canceled). This is different to the current behavior on Travis where failures are ignored. I suggest to try this configuration for a while (we are running in parallel with Travis anyway) to see if it is useful in practice. It should speed up CI times even more.

Reply:
That's a thing I actually wish Travis would allow us to do...